lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -604,9 +608,32 @@ newDBLayer trace defaultFieldValues mDatabaseFile = do
         , putDelegationCertificate = \(PrimaryKey wid) cert sl -> ExceptT $ do
             selectWallet wid >>= \case
                 Nothing -> pure $ Left $ ErrNoSuchWallet wid
-                Just _  -> pure <$> repsert
-                    (DelegationCertificateKey wid sl)
-                    (DelegationCertificate wid sl (W.dlgCertPoolId cert))
+                Just _  -> case cert of
+                    W.CertDelegateNone _ -> do
+                        repsert
+                            (DelegationCertificateKey wid sl)
+                            (DelegationCertificate wid sl Nothing)
+                        pure <$> repsert
+                            (StakeKeyCertificateKey wid sl)
+                            (StakeKeyCertificate wid sl False)
+                    W.CertDelegateFull _ pool ->
+                        pure <$> repsert
+                            (DelegationCertificateKey wid sl)
+                            (DelegationCertificate wid sl (Just pool))
+                    W.CertRegisterKey _ ->
+                        pure <$> repsert
+                            (StakeKeyCertificateKey wid sl)
+                            (StakeKeyCertificate wid sl True)
```

Comment:
Hmmm. This isn't quite right. Stake key are registered through their own certificates. So this is what we should be tracking. In particular, the case of ITN, the key is registered by _another mean_ than the wallet backend so we never get to create a registration certificate in the first place.

Reply:
Okay, went through too quickly at first when I saw the `repsert` and the boolean. Looks like this would work indeed and actually play nicely with both Jörmungandr and cardano-node.